### PR TITLE
Allow live feed file directory selection

### PR DESCRIPTION
**Pull request by tsingh777**
Fixes https://github.com/KhalisFoundation/sttm-desktop/issues/184
image

### CODE REVIEW

File: www/js/controller.js [hunk 1]
```diff
@@ -493,4 +494,20 @@ module.exports = {
...
+  livefeed(val) {
+    if (val) {
+      dialog.showOpenDialog({
+        defaultPath: remote.app.getPath('desktop'),
+        properties: ['openDirectory'],
+      }, (path) => {
+        store.set('userPrefs.app.live-feed-location', path[0]);
+        const locationLabel = document.getElementsByClassName('sub-label livefeed');
+
+        for (let i = 0, len = locationLabel.length; i < len; i += 1) {
```
Comment (navdeepsinghkhalsa) [inline www/js/controller.js]: How many places is it supposed to update? Is a `for` loop necessary?

Comment (tsingh777) [reply www/js/controller.js]: At the moment 2 places (display tab and setting pane), when https://github.com/KhalisFoundation/sttm-desktop/issues/280 is done, it should be one place, can refactor once that change is in. 

for loop might be much but it gets the job done.

File: app.js [hunk 2]
```diff
@@ -293,7 +293,8 @@ ipcMain.on('clear-apv', () => {
 });
 
 function createBroadcastFiles(arg) {
-  const userDataPath = electron.app.getPath('userData');
+  const liveFeedLocation = store.get('userPrefs.app.live-feed-location');
+  const userDataPath = (liveFeedLocation === 'default' || !liveFeedLocation) ? electron.app.getPath('desktop') : liveFeedLocation;
```
Comment (tsingh777) [inline app.js]: I hope `electron.app.getPath('desktop')` works for windows/linux as well. only tested on mac. 

cc. @navdeepsinghkhalsa

Comment (navdeepsinghkhalsa) [reply app.js]: It's officially documented, so they should be doing the legwork, here. I appreciated the find. Good work :)

**Pull request merged**